src/main/java/seedu/address/model/entry/Event.java

```diff
@@ -0,0 +1,121 @@
+package seedu.address.model.entry;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents an Event in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Event extends Entry {
+
+    //Identity fields
+    private final String company;
+
+    // Data fields
+    private final Date date;
+    private final Time time;
+    private final Platform platform;
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Event(Name name, String company, Date date, Time time, Platform platform, Set<Tag> tags) {
+        super(name, tags);
+        requireAllNonNull(company, date, time, platform);
+        this.company = company;
+        this.date = date;
+        this.time = time;
+        this.platform = platform;
+    }
+
+    public String getCompany() {
+        return this.company;
+    }
+
+    public Date getDate() {
+        return this.date;
+    }
+
+    public Time getTime() {
+        return this.time;
+    }
+
+    public Platform getPlatform() {
+        return this.platform;
+    }
+
+    /**
+     * Returns true if both entries are an event and have the same name.
+     * This defines a weaker notion of equality between two events.
+     */
+    @Override
+    public boolean isSameEntry(Entry otherEntry) {
+        if (otherEntry == this) {
+            return true;
+        }
+
+        if (!(otherEntry instanceof Event)) {
+            return false;
+        }
+
+        Event otherEvent = (Event) otherEntry;
+        return otherEvent.getName().equals(getName())
+                && otherEvent.getCompany().equals(getCompany());
```

Comment:
When checking isSameEntry, would it be better to check for having the same date and time as well? Since there can be instances where one might have multiple events of the same type at the same company, but they take place at different times.